### PR TITLE
Named tensor support for: atan2, output_nr, detach{_}, requires_grad_

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26543 Named tensor support for: atan2, output_nr, detach{_}, requires_grad_**

Also adds a test for logical_xor (it already had named tensor support
but there was no test)

Test Plan:
- [namedtensor ci]

Differential Revision: [D17501403](https://our.internmc.facebook.com/intern/diff/D17501403)

### CODE REVIEW

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -4899,9 +4903,11 @@
     CUDA: legacy::cuda::_th_dist
 
 - func: atan2.out(Tensor self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)
+  supports_named_tensor: True
```
Comment (nairbv) [inline aten/src/ATen/native/native_functions.yaml]: do we need any new named tensor tests for these?

**Review by nairbv — APPROVED** [review]
